manifests/composer.pp

```diff
@@ -42,15 +42,12 @@
   validate_bool($auto_update)
   validate_re("x${max_age}", '^x\d+$')
 
-  if $manage_curl { ensure_packages(['curl']) }
-
-  exec { 'download composer':
-    command     => "curl -L ${source} -o ${path}",
-    environment => $environment,
-    creates     => $path,
-    path        => ['/bin/', '/sbin/' , '/usr/bin/', '/usr/sbin/',
-                    '/usr/local/bin', '/usr/local/sbin'],
-    require     => [Class['::php::cli'],Package['curl']],
+  archive { 'download composer':
+    source       => $source,
+    creates      => $path,
+    proxy_type   => $proxy_type,
+    proxy_server => $proxy_server,
+    require      => Class['::php::cli'],
```

Comment:
This dependency looks a bit odd.  But it was there before, so maybe removing it will have unintended consequences.